samples/kvssink_gstreamer_sample.cpp

```diff
@@ -0,0 +1,845 @@
+    // index 1 is stream name which is already processed
+    for (int i = 2; i < argc; i++) {
+        if (i < argc) {
+            if ((0 == STRCMPI(argv[i], "-w")) ||
+                (0 == STRCMPI(argv[i], "/w")) ||
+                (0 == STRCMPI(argv[i], "--w"))) {
+                // process the width
+                if (STATUS_FAILED(STRTOI32(argv[i + 1], NULL, 10, &width))) {
+                    return 1;
+                }
+            } else if ((0 == STRCMPI(argv[i], "-h")) ||
+                       (0 == STRCMPI(argv[i], "/h")) ||
+                       (0 == STRCMPI(argv[i], "--h"))) {
+                // process the height
+                if (STATUS_FAILED(STRTOI32(argv[i + 1], NULL, 10, &height))) {
+                    return 1;
+                }
+            } else if ((0 == STRCMPI(argv[i], "-f")) ||
+                       (0 == STRCMPI(argv[i], "/f")) ||
+                       (0 == STRCMPI(argv[i], "--f"))) {
+                // process the framerate
+                if (STATUS_FAILED(STRTOI32(argv[i + 1], NULL, 10, &framerate))) {
+                    return 1;
+                }
+            } else if ((0 == STRCMPI(argv[i], "-b")) ||
+                       (0 == STRCMPI(argv[i], "/b")) ||
+                       (0 == STRCMPI(argv[i], "--b"))) {
+                // process the bitrate
+                if (STATUS_FAILED(STRTOI32(argv[i + 1], NULL, 10, &bitrateInKBPS))) {
+                    return 1;
+                }
+             } else if ((0 == STRCMPI(argv[i], "-runtime")) ||
+                       (0 == STRCMPI(argv[i], "/runtime")) ||
+                       (0 == STRCMPI(argv[i], "--runtime"))) {
+                // process the max runtime
+                if (STATUS_FAILED(STRTOI32(argv[i + 1], NULL, 10, &(data->max_runtime)))) {
+                    return 1;
+                }
+            // skip the index
+            }
+            i++;
+        } else if (0 == STRCMPI(argv[i], "-?") ||
+                   0 == STRCMPI(argv[i], "--?") ||
+                   0 == STRCMPI(argv[i], "--help")) {
+            g_printerr("Invalid arguments\n");
+            return 1;
+        } else if (argv[i][0] == '/' ||
+                   argv[i][0] == '-') {
+            // Unknown option
+            g_printerr("Invalid arguments\n");
+            return 1;
+        }
+    }
+
+    if ((width == 0 && height != 0) || (width != 0 && height == 0)) {
+        g_printerr("Invalid resolution\n");
+        return 1;
+    }
+
+    if(framerate <= 0 || bitrateInKBPS <= 0) {
+       g_printerr("Invalid input arguments\n");
+       return 1;
+    }
+
+    LOG_DEBUG("Streaming with live source and width: " << width << ", height: " << height << ", fps: " << framerate
+                                                       << ", bitrateInKBPS" << bitrateInKBPS);
+
+    GstElement *source_filter, *filter, *kvssink, *h264parse, *encoder, *source, *video_convert;
+
+    /* create the elemnents */
```

Comment:
let us  remove the launch sample cmd as we are not using file sink here

Reply:
Done